streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorTopology.java

```diff
@@ -59,4 +81,42 @@ public SourceNode source(String topic) {
         return stateStoreSuppliers;
     }
 
+    private String childrenToString(List<ProcessorNode<?, ?>> children) {
+        if (children == null || children.isEmpty()) {
+            return "";
+        }
+
+        StringBuilder sb = new StringBuilder("children [");
+        for (ProcessorNode child : children) {
+            sb.append(child.name() + ",");
+        }
+        sb.setLength(sb.length() - 1);
+        sb.append("]\n");
+
+        // recursively print children
+        for (ProcessorNode child : children) {
+            sb.append("\t\t\t" + child.toString());
+            if (!sinkNameToTopic.containsKey(child.name())) {
```

Comment:
It does get triggered, like below. In this case the sink node is the child of the current node and it's `toString` method gets called. The check later is whether to recursively print it's children too, which we skip:
`for (ProcessorNode child : children) {
            sb.append("\t\t\t" + child.toString());`